File: sdks/ts/src/sdk/models/operations/getfilesidthumbnailid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum GetFilesIdThumbnailIdExtensionEnum {
    Png = "png",
    Jpg = "jpg"
}


export class GetFilesIdThumbnailIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=extension" })
  extension: GetFilesIdThumbnailIdExtensionEnum;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_id" })
  fileId: string;
}


export class GetFilesIdThumbnailIdQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=max_height" })
  maxHeight?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=max_width" })
  maxWidth?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=min_height" })
  minHeight?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=min_width" })
  minWidth?: number;
}


export class GetFilesIdThumbnailIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetFilesIdThumbnailIdPathParams;

  @SpeakeasyMetadata()
  queryParams: GetFilesIdThumbnailIdQueryParams;
}


export class GetFilesIdThumbnailIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  headers: Record<string, string[]>;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getFilesIdThumbnailId200ImageJpgBinaryString?: Uint8Array;

  @SpeakeasyMetadata()
  getFilesIdThumbnailId200ImagePngBinaryString?: Uint8Array;
}

File: go-sdk/pkg/models/operations/getfilesidthumbnailid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetFilesIDThumbnailIDExtensionEnum string

const (
	GetFilesIDThumbnailIDExtensionEnumPng GetFilesIDThumbnailIDExtensionEnum = "png"
	GetFilesIDThumbnailIDExtensionEnumJpg GetFilesIDThumbnailIDExtensionEnum = "jpg"
)

type GetFilesIDThumbnailIDPathParams struct {
	Extension GetFilesIDThumbnailIDExtensionEnum `pathParam:"style=simple,explode=false,name=extension"`
	FileID    string                             `pathParam:"style=simple,explode=false,name=file_id"`
}

type GetFilesIDThumbnailIDQueryParams struct {
	MaxHeight *int64 `queryParam:"style=form,explode=true,name=max_height"`
	MaxWidth  *int64 `queryParam:"style=form,explode=true,name=max_width"`
	MinHeight *int64 `queryParam:"style=form,explode=true,name=min_height"`
	MinWidth  *int64 `queryParam:"style=form,explode=true,name=min_width"`
}

type GetFilesIDThumbnailIDRequest struct {
	Retries     *utils.RetryConfig
	PathParams  GetFilesIDThumbnailIDPathParams
	QueryParams GetFilesIDThumbnailIDQueryParams
}

type GetFilesIDThumbnailIDResponse struct {
	ClientError                                  *shared.ClientError
	ContentType                                  string
	Headers                                      map[string][]string
	StatusCode                                   int64
	GetFilesIDThumbnailID200ImageJpgBinaryString []byte
	GetFilesIDThumbnailID200ImagePngBinaryString []byte
}
